debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java

```diff
@@ -107,78 +86,154 @@
+            return 0;
+        }
+
+        if (!Objects.equals(keyFieldRegex, "") && keyFieldReplacement == null) {
+            problems.accept(
+                    KEY_FIELD_REPLACEMENT,
+                    null,
+                    String.format("%s must be non-null if %s is set.",
+                            KEY_FIELD_REPLACEMENT.name(),
+                            KEY_FIELD_REGEX.name()
+                    )
+            );
+            return 1;
+        }
+        return 0;
+    }
 
     @Override
     public void configure(Map<String, ?> props) {
         Configuration config = Configuration.from(props);
-        final Field.Set configFields = Field.setOf(TOPIC_REGEX, TOPIC_REPLACEMENT, KEY_FIELD_REGEX,
-                KEY_FIELD_REPLACEMENT);
+        final Field.Set configFields = Field.setOf(
+                TOPIC_REGEX,
+                TOPIC_REPLACEMENT,
+                KEY_FIELD_REGEX,
+                KEY_FIELD_REPLACEMENT
+        );
+
         if (!config.validateAndRecord(configFields, logger::error)) {
             throw new ConnectException("Unable to validate config.");
         }
 
-        topicRegex = Pattern.compile(config.getString(TOPIC_REGEX));
-        topicReplacement = config.getString(TOPIC_REPLACEMENT);
+        topicRegex = Pattern.compile(config.getString(TOPIC_REGEX.name()));
+        topicReplacement = config.getString(TOPIC_REPLACEMENT.name());
 
-        String keyFieldRegexString = config.getString(KEY_FIELD_REGEX);
+        String keyFieldRegexString = config.getString(KEY_FIELD_REGEX.name());
         if (keyFieldRegexString != null) {
             keyFieldRegexString = keyFieldRegexString.trim();
         }
-        if (keyFieldRegexString != null && keyFieldRegexString != "") {
-            keyFieldRegex = Pattern.compile(config.getString(KEY_FIELD_REGEX));
-            keyFieldReplacement = config.getString(KEY_FIELD_REPLACEMENT);
+        if (keyFieldRegexString != null && !Objects.equals(keyFieldRegexString, "")) {
+            keyFieldRegex = Pattern.compile(config.getString(KEY_FIELD_REGEX.name()));
+            keyFieldReplacement = config.getString(KEY_FIELD_REPLACEMENT.name());
         }
-        keyFieldName = config.getString(KEY_FIELD_NAME);
-
-        keySchemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
-        envelopeSchemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
+        keyFieldName = config.getString(KEY_FIELD_NAME.name());
     }
 
     @Override
     public R apply(R record) {
         final String oldTopic = record.topic();
         final String newTopic = determineNewTopic(oldTopic);
+
         if (newTopic == null) {
             return record;
         }
 
-        logger.debug("Applying topic name transformation from " + oldTopic + " to " + newTopic + ".");
+        if (logger.isDebugEnabled()) {
+            logger.debug("Applying topic name transformation from {} to {}", oldTopic, newTopic);
+        }
```

Comment:
Good change to parameterize the log statement rather than build up a string. However, the only time it is advantageous to wrap a log statement in an if-enabled check is if computing the parameters to the log statement are not trivial. But they are trivial in this case, so no if-enabled check is required.